lib/entities/app-definition.ts

```diff
@@ -0,0 +1,155 @@
+import cloneDeep from 'lodash/cloneDeep'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import { MetaSysProps, DefaultElements, CollectionProp } from '../common-types'
+import enhanceWithMethods from '../enhance-with-methods'
+import { createUpdateEntity, createDeleteEntity } from '../instance-actions'
+import { AxiosInstance } from 'axios'
+
+const entityPath = 'app_definitions'
+
+type Field =
+  | 'Symbol'
+  | 'Text'
+  | 'RichText'
+  | 'Integer'
+  | 'Number'
+  | 'Date'
+  | 'Location'
+  | 'Boolean'
+  | 'Object'
+
+type LinkType = 'Asset' | 'Entry'
+
+// Fields Types
+interface SingleFieldType {
+  type: Field
+}
+
+interface LinkFieldType {
+  type: 'Link'
+  linkType: LinkType
+}
+
+interface ArrayFieldType {
+  type: 'Array'
+  items: SingleFieldType | LinkFieldType
+}
+
+type FieldType = SingleFieldType | LinkFieldType | ArrayFieldType
+
+// Locations
+type AppLocation = 'app-config' | 'entry-sidebar' | 'entry-editor' | 'dialog'
```

Comment:
this definition of `AppLocation` options looks incomplete :thinking: there's also location 'page' now https://www.contentful.com/developers/docs/extensibility/app-framework/locations/

Reply:
good spot, I copied from `typings`, so it was missing from the beginning :) 

I'll add it.